source/azure_iot_hub_client.c

```diff
@@ -831,7 +924,15 @@ AzureIoTHubClientResult_t AzureIoTHubClient_SendTelemetry( AzureIoTHubClient_t *
             AZLogError( ( "Failed to publish telemetry: MQTT error=0x%08x", xMQTTResult ) );
             xResult = eAzureIoTHubClientPublishFailed;
         }
-        else
+
+        if( ( xMQTTResult == eAzureIoTMQTTSuccess ) && ( xQOS == eAzureIoTHubMessageQoS1 ) )
```

Comment:
else if ( ( xQOS == eAzureIoTHubMessageQoS1 ) && (xResult =prvWaitForPubAck() ) != eAzureIoTHubClientSuccess )
{
Log_Error("Failed to wait for pubACK")
}
else
{
   AZLogInfo( ( "Successfully sent telemetry message" ) );
   xResult = eAzureIoTHubClientSuccess;
}